services/patient.services.js

```diff
@@ -80,4 +117,30 @@ const getAllPatients = async ({ search, page, pageSize }) => {
   }
 };
 
-module.exports = { createPatient, getAllPatients };
+const getPatient = async ({ slug }) => {
+  try {
+    const patient = await db.Patient.findOne({
+      where: { slug },
+      attributes: ["id", "slug", "image", "firstName", "lastName", "dateOfBirth",
+        "email", "phoneNumber", "address", "city", "createdAt", "updatedAt"],
+    });
+    return patient;
+  } catch (err) {
+    throw new MedicaError("Unable to return patient");
+  }
+};
+
+const deletePatient = async ({ slug }) => {
```

Comment:
Since you have patient `id` on the UI you should update and delete patients by id instead of slug.